panoramapublic/src/org/labkey/panoramapublic/query/ExperimentAnnotationsTableInfo.java

```diff
@@ -724,4 +747,47 @@ String getRenderId()
             return "input-picker-div-instrument";
         }
     }
+
+    public static class CatalogEntryIconColumn extends DataColumn
+    {
+        public CatalogEntryIconColumn(ColumnInfo col)
+        {
+            super(col);
+            super.setCaption("Catalog Entry");
+        }
+
+        @Override
+        public void renderGridCellContents(RenderContext ctx, Writer out) throws IOException
+        {
+            User user = ctx.getViewContext().getUser();
+            if (user == null || user.isGuest())
+            {
+                HtmlString.NBSP.appendTo(out);
+                return;
+            }
+            Integer catalogEntryId = ctx.get(getColumnInfo().getFieldKey(), Integer.class);
+
+            // Get the experiment connected with this catalog entry.
+            Integer experimentId = ctx.get(FieldKey.fromParts("id"), Integer.class);
+            if (experimentId != null)
+            {
+                ExperimentAnnotations expAnnot = ExperimentAnnotationsManager.get(experimentId);
+                // Display the catalog entry link only if the user has the required permissions (Admin or PanoramaPublicSubmitter) in the the experiment folder.
+                if (expAnnot != null && CatalogEntryWebPart.canBeDisplayed(expAnnot, user))
+                {
+                    CatalogEntry entry = catalogEntryId == null ? null : CatalogEntryManager.get(catalogEntryId);
+                    String imageUrl = entry != null ? AppProps.getInstance().getContextPath() + "/PanoramaPublic/images/slideshow-icon-green.png"
+                                                    : AppProps.getInstance().getContextPath() + "/PanoramaPublic/images/slideshow-icon.png";
+                    String imageTitle = entry != null ? "View catalog entry" : "Add catalog entry";
+                    ActionURL catalogEntryLink = entry != null ? PanoramaPublicController.getViewCatalogEntryUrl(expAnnot, entry)
+                                                               : PanoramaPublicController.getAddCatalogEntryUrl(expAnnot).addReturnURL(ctx.getViewContext().getActionURL().clone());
+                    DOM.A(at(href, catalogEntryLink.getLocalURIString(), title, PageFlowUtil.filter(imageTitle)),
+                            DOM.IMG(at(src, imageUrl, height, 22, width, 22)))
+                            .appendTo(out);
+                    return;
+                }
+            }
+            HtmlString.EMPTY_STRING.appendTo(out);
```

Comment:
This codepath uses `EMPTY_STRING` and the one above uses `NBSP`. I imagine they should probably use the same placeholder, though maybe it doesn't make a difference.

Reply:
Done.